realm/realm-annotations-processor/src/main/java/io/realm/processor/Backlink.java

```diff
@@ -0,0 +1,247 @@
+import io.realm.annotations.Required;
+
+/**
+ * A Backlink is:
+ * <p>
+ * <code>
+ * class TargetClass {
+ *     // ...
+ *     {@literal @}LinkingObjects("sourceField")
+ *     RealmResults&lt;SourceClass&gt; targetField;
+ * }
+ * </code>.
+ * <p>
+ * The targetField of a managed object cannot be assigned.  It can be queried normally.
+ * When an instance X of a class with such a declaration is copied to Realm (`copyToRealm()`),
+ * the result will contain in its targetField references to any instances of SourceClass
+ * whose sourceField contains a reference to X.  Any previous contents of the targetField are lost.
+ * <p>
+ * When an instance X is copied from Realm (`copyFromRealm()`) the targetField in the returned object
+ * is just another field: it can be set normally. The field is initially set to be `null`.
+ * <p>
+ * Note that, because subclassing subclasses of RealmObject is forbidden, so are constructs like:
+ * <code>RealmResults&lt;? extends Foos&lt;</code>
+ * <p>
+ * In the code link direction is from the perspective of the link, not the backlink: the source is the
+ * instance to which the backlink points, the target is the instance holding the pointer.
+ * This is consistent with the use of terms in the Realm Core.
+ */
+final class Backlink {
+    private final VariableElement backlink;
+
+    /**
+     * The FQN of the class containing the field <code>targetField</code> that is
+     * annotated with the {@literal @}LinkingObjects annotation.
+     */
+    private final String targetClass;
+
+    /**
+     * The name of the backlinked field, in <code>targetClass</code>.
+     * A <code>RealmResults&lt;&gt;</code> field annotated with a {@literal @}LinkingObjects annotation.
+     */
+    private final String targetField;
+
+    /**
+     * The FQN of the class to which the backlinks, from <code>targetField</code>, point:
+     * The generic argument to the type of the <code>targetField</code> field.
+     */
+    private final String sourceClass;
+
+    /**
+     * The name of the field, in <code>sourceClass</code> that creates the backlink.
+     * Making this field, in an instance X of <code>sourceClass</code>,
+     * a reference to an instance Y of <code>targetClass</code>
+     * will cause the <code>targetField</code> of Y to contain a backlink to X.
+     */
+    private final String sourceField;
+
+
+    public Backlink(ClassMetaData klass, VariableElement backlink) {
+        if ((null == klass) || (null == backlink)) {
+            throw new NullPointerException(String.format("null parameter: %s, %s", klass, backlink));
+        }
+
+        this.backlink = backlink;
+        this.targetClass = klass.getFullyQualifiedClassName();
+        this.targetField = backlink.getSimpleName().toString();
+        this.sourceClass = getRealmResultsType(backlink);
+        this.sourceField = backlink.getAnnotation(LinkingObjects.class).value();
+    }
+
+    public String getTargetClass() { return targetClass; }
```

Comment:
Sure.  Lets just get the formatter fixed, so that I never have to get it wrong again, eh?